filemanager/digitaloceanmanager_v2.go

```diff
@@ -0,0 +1,328 @@
+package filemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go"
+	"github.com/mitchellh/mapstructure"
+	"github.com/samber/lo"
+
+	awsutil "github.com/rudderlabs/rudder-go-kit/awsutil_v2"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+type DigitalOceanConfigV2 struct {
+	Bucket         string
+	Prefix         string
+	EndPoint       string
+	AccessKeyID    string
+	AccessKey      string
+	Region         *string
+	ForcePathStyle *bool
+	DisableSSL     *bool
+}
+
+type DigitalOceanManagerV2 struct {
```

Comment:
what differs in digital ocean apart from how we create the client?

Reply:
It's literally the same. We could also do an implementation with digital ocean [SDK](https://github.com/digitalocean/godo) but deferred it for now